src/include/txn_inline.h

```diff
@@ -1450,6 +1451,23 @@ __wt_txn_read_upd_list_internal(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt,
 
             return (WT_PREPARE_CONFLICT);
         }
+
+        if (F_ISSET(upd, WT_UPDATE_RESTORED_FROM_DELTA) && upd->type == WT_UPDATE_STANDARD) {
+            WT_ASSERT(session, !F_ISSET(S2BT(session), WT_BTREE_IN_MEMORY));
+            /*
+             * If we see an update that is not visible to the reader and it is restored from delta,
+             * we should search the history store.
+             */
+            if (F_ISSET_ATOMIC_32(S2C(session), WT_CONN_HS_OPEN) &&
+              !F_ISSET(session->dhandle, WT_DHANDLE_HS)) {
```

Comment:
We should also check whether the dhandle is metadata. No need to search the hs for metadata.